File: src/main/java/io/github/aphelium/modules/coin/dtos/CreateFiatDTO.java
package io.github.aphelium.modules.coin.dtos;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import java.util.UUID;

@Data
public class CreateFiatDTO {

    @NotBlank(message = "The name is mandatory")
    private String name;
    @NotBlank(message = "The description is mandatory")
    private String description;
    @NotBlank(message = "The code is mandatory")
    private String code;
    private Boolean isActive;
    @NotBlank(message = "The country is mandatory")
    private String country;

}

File: src/main/kotlin/io/github/aphelium/modules/coin/dtos/CreateFiatDTO.kt
package io.github.aphelium.modules.coin.dtos

import javax.validation.constraints.NotBlank

class CreateFiatDTO {
    val name: @NotBlank(message = "The name is mandatory") String? = null
    val description: @NotBlank(message = "The description is mandatory") String? = null
    val code: @NotBlank(message = "The code is mandatory") String? = null
    val isActive: Boolean? = null
    val country: @NotBlank(message = "The country is mandatory") String? = null
}